### DISCUSSION TITLE
Ideal Training Settings

### DISCUSSION THREAD

**Discussion by ClaytonFarr** [cloneofsimo/lora#49] [Show and tell]
In lieu of a project wiki, perhaps we can collect and share here what settings are working best for folks.

I'm still trying to get my hands around things but these are ones I've seen shared with good results thus far (I'm sure there are others). @cloneofsimo, @nitrosocke, if I've confused any of the settings for what you've done, let me know.

| Training Goal               | Person (Unknown)                                          | Person (Known/Celeb)                                                      | Style                                                                     |
|:--------------------------- |:------------------------------------------------------------------- |:----------------------------------------------------------------------------- |:----------------------------------------------------------------------------- |
| By                          | xsteenbrugge                                                        | nitrosocke                                                                    | cloneofsimo                                                                   |
| Link                        | [link](https://twitter.com/xsteenbrugge/status/1602640284445204482) | [link](https://github.com/cloneofsimo/lora/issues/19#issuecomment-1347056011) | [link](https://github.com/cloneofsimo/lora/issues/19#issuecomment-1346943672) |
| Base Model                  | SD 2.1 base                                                         |                                                                               | SD2.1 base                                                                              |
| Resolution                  |                                                                     | 768                                                                           | 512                                                                              |
| Images                      | 20                                                                  | 5-10                                                                          | 10                                                                            |
| Captioned                   | Yes                                                                 | No                                                                            | No                                                                            |
| Steps                       | 400 / img                                                           | 100 / img                                                                     | 300 / img                                                                    |
| Unet Learning Rate          | 1-e4                                                                | 1-e4                                                                          | 1-e4                                                                          |
| Text Encoder Learning Rate  |                                                                     |                                                                               |                                                                               |
| Learning Rate Schedule      |                                                                     | constant                                                                      | constant                                                                              |
| Training Batch Size         |                                                                     | 4                                                                             |                                                                               |
| Gradient Accumulation Steps |                                                                     | 1                                                                             |   1                                                                            |
| Mixed Precision             |                                                                     | fp16                                                                          | fp16                                                                              |
| Use 8bit Adam               |                                                                     | Yes                                                                           |                                                                               |
| Prior Preservation          |                                                                     |                                                                               | false                                                                              |

### [What training settings are working best for you?](#)

**Comment by ClaytonFarr** [comment]


**Reply by brian6091** [reply]
Great idea! I suggest adding the base model (SDv1-4, 1-5, 2.1) since there seem to be some differences in how they fine-tune.

**Reply by ClaytonFarr** [reply]
Great point. When I’m back at computer I’ll look up to see if current examples mentioned model and add those that do.

**Comment by cloneofsimo** [comment]
This is great! I think when there is enough info on this, we can add this table to the readme as well.
